File: GameRanksClient/src/app/models/serverStructs/element/PublisherStruct.ts
import {Publisher} from "../../entity/Publisher";
import {GameStruct} from "./GameStruct";

export class PublisherStruct{
	publisher: Publisher;
	avgScore: number;
	rankTitle: string;
	games: GameStruct[];

	constructor(publisher: Publisher, avgScore: number, rankTitle: string, games: GameStruct[]){
		this.publisher = publisher;
		this.avgScore = avgScore;
		this.rankTitle = rankTitle;
		this.games = games;
	}
}

File: GameRanks/src/main/java/GameRanks/GameRanks/clientStruct/element/PublisherStruct.java
package GameRanks.GameRanks.clientStruct.element;

import GameRanks.GameRanks.model.Publisher;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class PublisherStruct {
    private Publisher publisher;
    public double avgScore;
    public String rankTitle;    //gold,silver,bronze,normal
    public Iterable<GameStruct> games;
}
